processor/spanmetricsprocessor/processor_test.go

```diff
@@ -255,18 +255,26 @@ func TestMetricKeyCache(t *testing.T) {
 
 	// Test
 	ctx := metadata.NewIncomingContext(context.Background(), nil)
-	err := p.ConsumeTraces(ctx, traces)
 
+	// 0 key was cached at beginning
+	assert.Equal(t, 0, len(p.metricKeyToDimensions.Keys()))
+	assert.Equal(t, 0, len(p.metricKeyToDimensions.evictedItems))
+
+	err := p.ConsumeTraces(ctx, traces)
 	// Validate
 	require.NoError(t, err)
+	// 2 key was cached, 1 key was evicted and cleaned after the processing
+	assert.Equal(t, metricKeyToDimensionsCacheSize, len(p.metricKeyToDimensions.Keys()))
+	assert.Equal(t, 0, len(p.metricKeyToDimensions.evictedItems))
 
-	origKeyCache := make(map[metricKey]pdata.AttributeMap)
-	for k, v := range p.metricKeyToDimensions {
-		origKeyCache[k] = v
-	}
+	// consume another batch of traces
 	err = p.ConsumeTraces(ctx, traces)
+	// 2 key was cached, other keys were evicted and cleaned after the processing
+	assert.Equal(t, metricKeyToDimensionsCacheSize, len(p.metricKeyToDimensions.Keys()))
+	assert.Equal(t, 0, len(p.metricKeyToDimensions.evictedItems))
+
+
 	require.NoError(t, err)
-	assert.Equal(t, origKeyCache, p.metricKeyToDimensions)
```

Comment:
I'm not sure if this previous test logic is needed. As `origKeyCache` is copied from `p.metricKeyToDimensions`, they should always be equal?